### ISSUE TITLE
Add $schema to schema

### ISSUE THREAD

**Issue by moltar** [DavidAnson/markdownlint#227]
Adding the "$schema" prop would allow IDE support for IntelliSense.

One can add it to the config file themselves, but then IDE throws a little warning that "Property $schema is not allowed." Not a big deal, but it'd be nice to do away without the warning.

```json
{
  "$schema": "./node_modules/markdownlint/schema/markdownlint-config-schema.json",
  "default": true,
  "MD013": {
    "line_length": 100
  }
}
```

**Comment by moltar** [comment]
<img width="433" alt="image" src="https://user-images.githubusercontent.com/491247/66706803-bd6a7200-ed05-11e9-807c-109d9eeb5370.png">

**Comment by DavidAnson** [comment]
VS Code with the `markdownlint` extension installed should already offer schema-based hints and completion without any changes required. What editor are you using above? And what would be necessary to prevent it from complaining about the `$schema` property?

**Comment by DavidAnson** [comment]
What I’m finding on the web suggests that the `$schema` keyword only belongs in the schema itself, and not in the data - which is what `.markdownlint.json` represents in your example above.

https://json-schema.org/understanding-json-schema/basics.html

If you could please include a link to some documentation explaining the scenario you show, that would be greatly appreciated!

**Comment by moltar** [comment]
Maybe it's just VS Code specific, but it is documented here, and I can confirm it works.

https://code.visualstudio.com/docs/languages/json#_json-schemas-and-settings

Even without the extensions, I get IntelliSense.

Here's a video demo of that: https://drive.google.com/file/d/14TvQkNnwSFo80upePXA-oUudln-qIWMP/view?usp=drivesdk

I renamed `$schema` to `_$schema` first to disable the behavior, which then produces meaningless IntelliSense suggestions. After I renamed it back, the IS came back.
